### PR TITLE
Chore: Set tsconfig target to ES6

### PR DESCRIPTION
**Pull request by jackw**
**What this PR does / why we need it**:
This PR bumps the `@grafana/tsconfig` dependency to 1.2.0-rc1. This bump sets the [typescript target](https://www.typescriptlang.org/tsconfig#target) to `es6` for jest environments and our NPM package compilation.

Note: Due to toolkit using Webpack 4 and its `webpack.config` plugins built with toolkit are still compiled down to es5 compatible code.

**Which issue(s) this PR fixes**:

Related #44738

**Special notes for your reviewer**:

I'm still slightly unsure of the impact this will have but have tested the app, and published the packages locally then built  couple of plugins and run their tests (both unit and e2e) and it doesn't seem to introduce any issues anywhere other than in the Grafana tests. 🤞

### CODE REVIEW

File: public/app/features/explore/Wrapper.tsx [hunk 1]
```diff
@@ -53,8 +53,6 @@ class WrapperUnconnected extends PureComponent<Props> {
       locationService.partial({ from: undefined, to: undefined }, true);
     }
 
-    const richHistory = getRichHistory();
```
[the remark this answers is not in the document]
Comment (ifrost) [reply public/app/features/explore/Wrapper.tsx:56]: Should be fixed by https://github.com/grafana/grafana/pull/44975

**Review by leventebalogh — APPROVED** [review]
LGTM 👍

**Review by mckn — APPROVED** [review]
LGTM!

**Review by dprokop — APPROVED** [review]
LGTM

**Pull request merged**